contracts/factory/DAOFactory.sol

```diff
@@ -1,33 +1,46 @@
 pragma solidity 0.4.24;
 
+import "../acl/IACL.sol";
+import "../acl/ACL.sol";
 import "../kernel/IKernel.sol";
 import "../kernel/Kernel.sol";
 import "../kernel/KernelProxy.sol";
-
-import "../acl/IACL.sol";
-import "../acl/ACL.sol";
-
+import "../kill_switch/KillSwitch.sol";
+import "../kill_switch/IssuesRegistry.sol";
 import "./EVMScriptRegistryFactory.sol";
 
 
 contract DAOFactory {
+    string private constant ERROR_MISSING_BASE_KILL_SWITCH = "DF_MISSING_BASE_KILL_SWITCH";
+
     IKernel public baseKernel;
     IACL public baseACL;
-    EVMScriptRegistryFactory public regFactory;
+    KillSwitch public baseKillSwitch;
+    EVMScriptRegistryFactory public scriptsRegistryFactory;
 
     event DeployDAO(address dao);
-    event DeployEVMScriptRegistry(address reg);
+    event DeployEVMScriptRegistry(address registry);
 
     /**
-    * @notice Create a new DAOFactory, creating DAOs with Kernels proxied to `_baseKernel`, ACLs proxied to `_baseACL`, and new EVMScriptRegistries created from `_regFactory`.
+    * @notice Create a new DAOFactory, creating DAOs with Kernels proxied to `_baseKernel`, ACLs proxied to `_baseACL`, and new EVMScriptRegistries created from `_scriptsRegistryFactory`.
     * @param _baseKernel Base Kernel
     * @param _baseACL Base ACL
-    * @param _regFactory EVMScriptRegistry factory
+    * @param _scriptsRegistryFactory EVMScriptRegistry factory
     */
-    constructor(IKernel _baseKernel, IACL _baseACL, EVMScriptRegistryFactory _regFactory) public {
+    constructor(
+        IKernel _baseKernel,
+        IACL _baseACL,
+        KillSwitch _baseKillSwitch,
+        EVMScriptRegistryFactory _scriptsRegistryFactory
+    )
+        public
+    {
         // No need to init as it cannot be killed by devops199
-        if (address(_regFactory) != address(0)) {
-            regFactory = _regFactory;
+        if (address(_scriptsRegistryFactory) != address(0)) {
+            scriptsRegistryFactory = _scriptsRegistryFactory;
+        }
+        if (address(_baseKillSwitch) != address(0)) {
+            baseKillSwitch = _baseKillSwitch;
         }
 
         baseKernel = _baseKernel;
```

Comment:
The problem is that you won't notice any further changes in the contracts that implements those IDs in the future, we could cache it for a single transaction, but caching it "forever" seems a lot to me. Note that caching it for a single transaction will result in a lot of code duplication, it is the same trade-off we will always face, code simplicity vs. gas optimization. In this case I opted for the first one since it is a single call per DAO, but I'll try to optimize its costs as much as possible.

Reply:
> you won't notice any further changes in the contracts that implement those IDs

Yes, but in the case of `DAOFactory`, there's no way to change the base implementations we get of the `Kernel`, `ACL`, and `KillSwitch`. Outside of funny business with CREATE2, these IDs shouldn't change.

If we want to handle the possible CREATE2 scenario where these bases are replaced, then let's keep the code simpler and just query them when we need them 👍

Reply:
Anyway, I refactored it in a way we query roles only once per new DAO